virtual_ecosystem/models/abiotic/wind.py

```diff
@@ -302,27 +313,81 @@ def mix_and_ventilate(
     """
 
     input_variable_mixed = input_variable.copy()
-    for i in range(2, len(input_variable) - 1):
-        flux_up = (
-            mixing_coefficient[i - 1]
-            * (input_variable[i - 1] - input_variable[i])
-            / layer_thickness[i]
-        )
-        flux_down = (
-            mixing_coefficient[i + 1]
-            * (input_variable[i + 1] - input_variable[i])
-            / layer_thickness[i]
-        )
-        change_input_variable = (
-            (flux_up + flux_down) * time_interval / layer_thickness[i]
-        )
-        input_variable_mixed[i] += change_input_variable
-
-    # Ventilation at top
-    input_variable_mixed[0] += (
-        ventilation_rate * (input_variable[0] - input_variable[1]) * time_interval
-    )
-
+    n_layers, n_cells = input_variable.shape
+
+    for j in range(n_cells):
+        for i in range(1, n_layers - 1):  # skip top and bottom
+            center_val = input_variable[i, j]
+
+            if not np.isfinite(center_val):
+                continue
+
+            # ---- Upward flux ----
+            # Look for nearest valid input value above the current layer
+            upper_i = i - 1
+            while upper_i >= 0 and not np.isfinite(input_variable[upper_i, j]):
+                upper_i -= 1
+
+            # If a valid upper value and a mixing coefficient are found, compute flux
+            if upper_i >= 0 and np.isfinite(mixing_coefficient[upper_i, j]):
+                delta_up = input_variable[upper_i, j] - center_val
```

Comment:
Should one of `delta_up` or `delta_down` be reversed? Assuming a positive or negative gradient through the canopy, one of the `delta` values is going to yield negative values and then gets clipped by `flux_up < 0`

Reply:
I removed the clipping step to allow for flows in both directions